### ISSUE TITLE
Charmap codec can't encode

### ISSUE THREAD

**Issue by selinakvle** [CorentinJ/Real-Time-Voice-Cloning#89]
Hi,

I'm running the cli demo and I got all the way to "Write a sentence.." and I get this error.  Could you please help?  I've been trying to get this to work for me since 8/8 and working through many setbacks...I'm finally close.

`
Created the mel spectrogram
Synthesizing the waveform:
Traceback (most recent call last):
  File "demo_cli.py", line 161, in <module>
    generated_wav = vocoder.infer_waveform(spec)
  File "C:\Users\selinakvle\Real-Time-Voice-Cloning\vocoder\inference.py", line 57, in infer_waveform
    wav = _model.generate(mel, batched, target, overlap, hp.mu_law, progress_callback)
  File "C:\Users\selinakvle\Real-Time-Voice-Cloning\vocoder\models\fatchord_version.py", line 219, in generate
    progress_callback(i, seq_len, b_size, gen_rate)
  File "C:\Users\selinakvle\Real-Time-Voice-Cloning\vocoder\models\fatchord_version.py", line 248, in gen_display
    stream(msg)
  File "C:\Users\selinakvle\Real-Time-Voice-Cloning\vocoder\display.py", line 16, in stream
    sys.stdout.write("\r{%s}" % message)
  File "C:\Users\selinakvle\AppData\Local\Programs\Python\Python36\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode characters in position 4-19: character maps to <undefined>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "demo_cli.py", line 184, in <module>
    print("Caught exception: %s" % repr(e))
  File "C:\Users\selinakvle\AppData\Local\Programs\Python\Python36\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode characters in position 54-69: character maps to <undefined>
`

**Comment by CorentinJ** [comment]
Are you writing your sentences in plain english characters?

**Comment by blue-fish** [comment]
The progbar in vocoder/display.py is incompatible with the encoding specified for python stdout. It has length 16 and occupies position 4-19 of the vocoder message stream, consistent with the 1st error message. Someone also reported a similar issue in #50 .

**Comment by CorentinJ** [comment]
[This](https://github.com/CorentinJ/Real-Time-Voice-Cloning/blob/master/vocoder/display.py#L11)? They could simply be replaced then, or better, we'd use tqdm everywhere in the repo

**Comment by blue-fish** [comment]
Yes, that line of code. This is what it looks like using `#` and `-` to generate the progbar. Not too bad though I prefer how it looks currently.

```
Synthesizing the waveform:
{| #############--- 36000/48000 | Batch Size: 5 | Gen Rate: 0.9kHz | }
```

I already merged the workaround #372. Do you want to back it out and implement something like this instead? We could also use tqdm with ascii=True. However, everyone would get the ASCII progbar unless we can figure out a way to detect incompatible systems.

**Comment by CorentinJ** [comment]
It's fine, no need to get too fancy
